File: src/main/java/me/pcasaes/hexoids/infrastructure/index/FullScanPlayerSpatialIndex.java
package me.pcasaes.hexoids.infrastructure.index;

import io.quarkus.arc.DefaultBean;
import me.pcasaes.hexoids.core.domain.index.PlayerSpatialIndex;
import me.pcasaes.hexoids.core.domain.model.Game;
import me.pcasaes.hexoids.core.domain.model.Player;

import javax.enterprise.context.ApplicationScoped;
import javax.inject.Inject;
import java.util.logging.Logger;

@DefaultBean
@ApplicationScoped
public class FullScanPlayerSpatialIndex implements PlayerSpatialIndex {

    private static final Logger LOGGER = Logger.getLogger(FullScanPlayerSpatialIndex.class.getName());

    @Inject
    public FullScanPlayerSpatialIndex() {
        LOGGER.info("Using Full Scan Spatial Index");
    }

    @Override
    public Iterable<Player> search(float x1, float y1, float x2, float y2, float distance) {
        return Game.get().getPlayers();
    }
}

File: src/main/kotlin/me/pcasaes/hexoids/infrastructure/index/FullScanPlayerSpatialIndex.kt
package me.pcasaes.hexoids.infrastructure.index

import io.quarkus.arc.DefaultBean
import jakarta.enterprise.context.ApplicationScoped
import jakarta.inject.Inject
import me.pcasaes.hexoids.core.domain.index.PlayerSpatialIndex
import me.pcasaes.hexoids.core.domain.model.Game
import me.pcasaes.hexoids.core.domain.model.Player
import java.util.logging.Logger

@DefaultBean
@ApplicationScoped
class FullScanPlayerSpatialIndex @Inject constructor() : PlayerSpatialIndex {
    init {
        LOGGER.info("Using Full Scan Spatial Index")
    }

    override fun search(x1: Float, y1: Float, x2: Float, y2: Float, distance: Float): Iterable<Player> {
        return Game.get().getPlayers()
    }

    companion object {
        private val LOGGER: Logger = Logger.getLogger(FullScanPlayerSpatialIndex::class.java.getName())
    }
}
